examples/user_interfaces/embedding_in_tk_sgskip.py

```diff
@@ -30,18 +32,39 @@
 toolbar = NavigationToolbar2Tk(canvas, root, pack_toolbar=False)
 toolbar.update()
 
-
 canvas.mpl_connect(
     "key_press_event", lambda event: print(f"you pressed {event.key}"))
 canvas.mpl_connect("key_press_event", key_press_handler)
 
-button = tkinter.Button(master=root, text="Quit", command=root.quit)
+button_quit = tkinter.Button(master=root, text="Quit", command=root.quit)
+
+
+def update_frequency(new_val):
+    # retrieve frequency
+    f = float(new_val)
+
+    # retrieve Axes object and update plots there
+    ax = fig.axes[0]
+    # alternatively you could delete last plotted line `del ax.lines[0]`
+    ax.clear()
+
+    ax.plot(t, 2 * np.sin(2 * np.pi * f * t))
+    #  since Axes was cleared
+    ax.set_xlabel("x")
```

Comment:
There's likely no need to clear and re-plot everything. Save the result of `ax.plot` and call `.set_data` on it. Or if that's not something you want to show, then call `.remove` on it.

Reply:
I will implement that for this example.